src/libs/TransactionUtils.ts

```diff
@@ -307,11 +308,30 @@ function getOriginalAmount(transaction: Transaction): number {
     return Math.abs(amount);
 }
 
+/**
+ * Verify if the transaction is of Distance request and is not fully ready:
+ * - it has a zero amount, which means the request was created offline and expects the distance calculation from the server
```

Comment:
I'm wondering if you think we should change this created offline flow so that it also sets the `isLoading` state to `true`. That would simplify the logic here so that it's always based on the `isLoading` state. I'm not sure what other implications that has for where `isLoading` is used though.

It just seems very nonintuitive that a zero amount means it was created offline and it's waiting for a server response.

Reply:
At first, I intended to implement this change as well: set `transaction.isLoading` also when creating the Distance request. There were some complications and I decided it was not worth the struggle as it was _a bit_ out of scope.
I can give it another shot though 🤔

Reply:
Just to clarify: we should set `isLoading` only on the Distance requests that are created offline?
For those created online, the amount and distance are set properly and we don't want to show `TBD`, but the map snapshot is technically also "loading" from the BE:

<img width="384" alt="image" src="https://github.com/Expensify/App/assets/12595293/a8c4b9b9-606f-43dd-818f-77d20e0007c5">

Reply:
> The string "TBD" I think should be set in the optimistic data, rather than calculated on the fly

The difficulty with this is that the `amount` has the `number` type, and we do some manipulations like `+amount` or `-amount` depending on the type of expense. It won't settle well with the amount optimistically set to `TBD` string.

Reply:
> `isLoading` is supposed to be used for the map/image
`isLoading` shouldn't be used for anything other than the image, so let's stop using it here

@tgolen  I don't find any evidence for these statements in the code.
It's used for disabling the "Next" or "Save" buttons when creating or updating the Distance request online:

https://github.com/Expensify/App/blob/4097c00a9c093253d62b3785da9f87cc4d2e8e46/src/components/DistanceRequest/index.js#L90

Reply:
I'm drawing that conclusion from two things:

1. The `isLoading` property is optimistically set based on the presence of waypoints ([here](https://github.com/Expensify/App/blob/9cf1ed840158a6e5e536a38b791be6cf39ae7416/src/libs/actions/IOU.js#L1029))
2. The original intention of when this was added, regardless of how it's been possibly misused in other areas

Reply:
Got you. [This concern](https://github.com/Expensify/App/pull/34135#discussion_r1455799535) is still relevant though:

>The difficulty with this is that the amount has the number type, and we do some manipulations like +amount or -amount depending on the type of expense. It won't settle well with the amount optimistically set to TBD string.

I'm talking about this function, for instance. Using `Math.abs('tbd')` will cause isNaN-related issues: https://github.com/Expensify/App/blob/4097c00a9c093253d62b3785da9f87cc4d2e8e46/src/libs/TransactionUtils.ts#L261-L282

Also, setting the "TBD" in the optimistic data will make it non-translatable, but it's a rather minor issue since the optimistic data remains only on the device.

@tgolen WDYT?

Reply:
Rather than storing "TBD" in the optimistic data, I think we should store the translated phrase. This would be slightly weird if they switch languages while offline, but I think that's too rare of an edge case to consider.

Reply:
Yes, that's what I meant – only changing language when offline will be affected.
But the non-numeric `amount` is more critical here.